drivers/UARTSerial.cpp

```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-#if DEVICE_SERIAL
+#if (DEVICE_SERIAL && DEVICE_INTERRUPTIN)
```

Comment:
Dude, my bad totally missed it. Definitely the right time to use the flag! Good catch.